protected/humhub/modules/comment/models/forms/CommentForm.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace humhub\modules\comment\models\forms;
+
+use humhub\modules\comment\models\Comment;
+use humhub\modules\content\components\ContentActiveRecord;
+use Yii;
+use yii\web\ServerErrorHttpException;
+
+/**
+ * CommentForm
+ * @package humhub\modules\comment\models\forms
+ *
+ * @since 0.5
+ */
+class CommentForm extends yii\base\Model
+{
+    /**
+     * Comment message
+     * @var string
+     */
+    public $message;
+
+    /**
+     * The list of files attached to a comment
+     * @var array
+     */
+    public $fileList;
+
+    /**
+     * @var Comment The model to comment
+     */
+    public $comment;
+
+    /**
+     * @var Comment|ContentActiveRecord The model to comment or other active content
+     */
+    public $target;
+
+    public function __construct($target, $comment)
+    {
+        $this->target = $target;
+        $this->comment = $comment;
+        parent::__construct();
+    }
+
+    /**
+     * @return array validation rules for model attributes.
+     */
+    public function rules()
+    {
+        return [
+            [['message'], 'required', 'isEmpty' => function ($message) {
+                $hasFile = !empty($this->fileList) || (!$this->comment->isNewRecord && $this->comment->fileManager->find()->count());
+
+                //check if message empty and attached file exists
+                if (empty($message) && !$hasFile) {
+                    throw new ServerErrorHttpException(Yii::t('CommentModule.base', 'The comment must not be empty!'));
+                }
+            }],
+            [['fileList'], 'safe'],
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function load($data, $formName = null)
+    {
+        // When user updates comment $data contain 'Comment', otherwise not
+        if (isset($data['Comment'])) {
+            $data['message'] = $data['Comment']['message'];
+            unset($data['Comment']);
+        }
+
+        if (!parent::load($data, $formName)) {
+            return false;
+        }
+
+        if (!$this->validate()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function formName()
+    {
+        return '';
+    }
+
+
+    /**
+     * Updates the form
+     *
+     * @return boolean
+     */
+    public function update()
+    {
+        $this->comment->message = $this->message;
+        return $this->save();
+    }
+
+
+    /**
+     * Saves the form
+     *
+     * @return boolean
+     */
+    public function save()
+    {
+        /**@var Comment $this->comment*/
+        $this->comment->setPolyMorphicRelation($this->target);
+
+        //check if model saved
+        if ($this->comment->save()) {
+            $this->comment->fileManager->attach($this->fileList);
+
+            // Reload comment to get populated created_at field
+            $this->comment->refresh();
```

Comment:
The refresh line can be removed, since it was fixed by another PR.

Reply:
Done!

Reply:
```
{
    "url": "/index.php?r=comment%2Fcomment%2Fedit&id=335&objectModel=humhub%5Cmodules%5Cpolls%5Cmodels%5CPoll&objectId=6",
    "status": 500,
    "response": "<pre>Exception &#039;yii\\base\\InvalidArgumentException&#039; with message &#039;&#039;NOW()&#039; is not a valid date time value: DateTime::__construct(): Failed to parse time string (NOW()) at position 3 ((): Unexpected character\nArray\n(\n    [warning_count] =&gt; 0\n    [warnings] =&gt; Array\n        (\n        )\n\n    [error_count] =&gt; 2\n    [errors] =&gt; Array\n        (\n            [3] =&gt; Unexpected character\n            [4] =&gt; Unexpected character\n        )\n\n)\n&#039; \n\nin /var/www/humh...",
    "textStatus": "error",
    "xhr": {
        "readyState": 4,
        "responseText": "<pre>Exception &#039;yii\\base\\InvalidArgumentException&#039; with message &#039;&#039;NOW()&#039; is not a valid date time value: DateTime::__construct(): Failed to parse time string (NOW()) at position 3 ((): Unexpected character\nArray\n(\n    [warning_count] =&gt; 0\n    [warnings] =&gt; Array\n        (\n        )\n\n    [error_count] =&gt; 2\n    [errors] =&gt; Array\n        (\n            [3] =&gt; Unexpected character\n            [4] =&gt; Unexpected character\n        )\n\n)\n&#039; \n\nin /var/www/humhub/protected/vendor/yiisoft/yii2/i18n/Formatter.php:867\n\nStack trace:\n#0 /var/www/humhub/protected/vendor/yiisoft/yii2/i18n/Formatter.php(761): yii\\i18n\\Formatter-&gt;normalizeDatetimeValue(Object(yii\\db\\Expression))\n#1 /var/www/humhub/protected/vendor/yiisoft/yii2/i18n/Formatter.php(724): yii\\i18n\\Formatter-&gt;formatDateTimeValue(Object(yii\\db\\Expression), &#039;medium&#039;, &#039;datetime&#039;)\n#2 /var/www/humhub/protected/humhub/modules/comment/widgets/views/comment.php(75): yii\\i18n\\Formatter-&gt;asDatetime(Object(yii\\db\\Expression))\n#3 /var/www/humhub/protected/vendor/yiisoft/yii2/base/View.php(348): require(&#039;/var/www/humhub...&#039;)\n#4 /var/www/humhub/protected/vendor/yiisoft/yii2/base/View.php(257): yii\\base\\View-&gt;renderPhpFile(&#039;/var/www/humhub...&#039;, Array)\n#5 /var/www/humhub/protected/vendor/yiisoft/yii2/base/View.php(156): yii\\base\\View-&gt;renderFile(&#039;/var/www/humhub...&#039;, Array, Object(humhub\\modules\\comment\\widgets\\Comment))\n#6 /var/www/humhub/protected/vendor/yiisoft/yii2/base/Widget.php(236): yii\\base\\View-&gt;render(&#039;comment&#039;, Array, Object(humhub\\modules\\comment\\widgets\\Comment))\n#7 /var/www/humhub/protected/humhub/modules/comment/widgets/Comment.php(58): yii\\base\\Widget-&gt;render(&#039;comment&#039;, Array)\n#8 /var/www/humhub/protected/humhub/components/Widget.php(63): humhub\\modules\\comment\\widgets\\Comment-&gt;run()\n#9 /var/www/humhub/protected/humhub/modules/comment/controllers/CommentController.php(157): humhub\\components\\Widget::widget(Array)\n#10 [internal function]: humhub\\modules\\comment\\controllers\\CommentController-&gt;actionEdit()\n#11 /var/www/humhub/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)\n#12 /var/www/humhub/protected/vendor/yiisoft/yii2/base/Controller.php(180): yii\\base\\InlineAction-&gt;runWithParams(Array)\n#13 /var/www/humhub/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\\base\\Controller-&gt;runAction(&#039;edit&#039;, Array)\n#14 /var/www/humhub/protected/vendor/yiisoft/yii2/web/Application.php(103): yii\\base\\Module-&gt;runAction(&#039;comment/comment...&#039;, Array)\n#15 /var/www/humhub/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\\web\\Application-&gt;handleRequest(Object(humhub\\components\\Request))\n#16 /var/www/humhub/index.php(25): yii\\base\\Application-&gt;run()\n#17 {main}</pre>",
        "status": 500,
        "statusText": "Internal Server Error"
    },
    "dataType": "html",
    "html": "<pre>Exception &#039;yii\\base\\InvalidArgumentException&#039; with message &#039;&#039;NOW()&#039; is not a valid date time value: DateTime::__construct(): Failed to parse time string (NOW()) at position 3 ((): Unexpected character\nArray\n(\n    [warning_count] =&gt; 0\n    [warnings] =&gt; Array\n        (\n        )\n\n    [error_count] =&gt; 2\n    [errors] =&gt; Array\n        (\n            [3] =&gt; Unexpected character\n            [4] =&gt; Unexpected character\n        )\n\n)\n&#039; \n\nin /var/www/humh...",
    "error": {},
    "errorThrown": "Internal Server Error",
    "validationError": false
}
```


When I removed the **$this->comment->refresh();** . I have got this error. Maybe in my branch right now this issue has nor resolved yet. Keep it without **$this->comment->refresh();**?

Reply:
Hmm, then leave it there, we can remove it afterwards.

Reply:
Ok!